pkg/services/ngalert/ngalert.go

```diff
@@ -717,6 +732,40 @@ func configureHistorianBackend(
 	return nil, fmt.Errorf("unrecognized state history backend: %s", backend)
 }
 
+func configureNotificationHistorian(
+	ctx context.Context,
+	featureToggles featuremgmt.FeatureToggles,
+	cfg setting.UnifiedAlertingNotificationHistorySettings,
+	met *metrics.NotificationHistorian,
+	l log.Logger,
+	tracer tracing.Tracer,
+) (nfstatus.NotificationHistorian, error) {
+	if !featureToggles.IsEnabled(ctx, featuremgmt.FlagAlertingNotificationHistory) {
+		return nil, nil
+	}
+
+	if !cfg.Enabled {
+		met.Info.Set(0)
+		return nil, nil
+	}
+
+	met.Info.Set(1)
```

Comment:
Double check the case of this metric in the case of it being enabled, the feature flag being turned off, and then grafana restarts.

Reply:
Good call! I added a call to `met.Info.Set(0)` when the feature toggle is disabled: [commit](https://github.com/grafana/grafana/pull/107644/commits/251eaa1dcdc5efda8513e69914889e29ec0f1ecb), so we don't have to guess